src/calculatePlayerMoves.ts

```diff
@@ -0,0 +1,243 @@
+  // When targeted square is occupied by player
+  if (square.playerId) return false;
+
+  // When wall to the side of player's origin square is not walkable
+  let coordinatesWallInBetween: Coordinates = {
+    y: startFromHorizontal
+      ? coordinatesPlayer.y
+      : coordinatesPlayer.y + (coordinatesSquare.y - coordinatesPlayer.y) / 2, // division by 2, because coordinate difference between player and targeted square is 2 and we select the middle square
+    x: startFromHorizontal ? coordinatesPlayer.x + (coordinatesSquare.x - coordinatesPlayer.x) / 2 : coordinatesPlayer.x
+  };
+  let wallInBetween = getSquareByCoordinates(coordinatesWallInBetween, board);
+  if (wallInBetween.type !== SquareType.Wall) throw new Error("Square is not of type 'Wall'");
+  if (!wallInBetween.isWalkable) return false;
+
+  // When player square to the side of player's origin is not other player
+  const coordinatesPlayerInBetween: Coordinates = {
+    y: startFromHorizontal ? coordinatesPlayer.y : coordinatesSquare.y,
+    x: startFromHorizontal ? coordinatesSquare.x : coordinatesPlayer.x
+  };
+  const playerInBetween = getSquareByCoordinates(coordinatesPlayerInBetween, board);
+  if (playerInBetween.type !== SquareType.Player) throw new Error("Square is not of type 'Player'");
+  if (!playerInBetween.playerId) return false;
+
+  // When wall between the targeted square and previous square is not walkable
+  coordinatesWallInBetween = {
+    y: startFromHorizontal
+      ? coordinatesPlayer.y + (coordinatesSquare.y - coordinatesPlayer.y) / 2 // division by 2, because coordinate difference between player and targeted square is 2 and we select the middle square
+      : coordinatesSquare.y,
+    x: startFromHorizontal ? coordinatesSquare.x : coordinatesPlayer.x + (coordinatesSquare.x - coordinatesPlayer.x) / 2
+  };
+  wallInBetween = getSquareByCoordinates(coordinatesWallInBetween, board);
+  if (wallInBetween.type !== SquareType.Wall) throw new Error("Square is not of type 'Wall'");
+  if (!wallInBetween.isWalkable) return false;
+
+  // When targeted square is not near the border of the board or there is no wall that would block linear jump
+  const LAST_ROW_INDEX = BOARD_WIDTH - 1;
+  if (
+    (startFromHorizontal ? coordinatesSquare.x : coordinatesSquare.y) !== 0 &&
+    (startFromHorizontal ? coordinatesSquare.x : coordinatesSquare.y) !== LAST_ROW_INDEX
+  ) {
+    coordinatesWallInBetween = {
+      y: startFromHorizontal
+        ? coordinatesPlayer.y
+        : coordinatesPlayer.y + ((coordinatesSquare.y - coordinatesPlayer.y) / 2) * 3, // division by 2 and multiplication by 3, because coordinate difference between player and targeted square is 2 and we select the square that is over the targeted square
+      x: startFromHorizontal
+        ? coordinatesPlayer.x + ((coordinatesSquare.x - coordinatesPlayer.x) / 2) * 3
+        : coordinatesPlayer.x
+    };
+    wallInBetween = getSquareByCoordinates(coordinatesWallInBetween, board);

```

Comment:
For this wall, just get the coordinates first, no matter if enemy is at the edge of the board and then check if wall is either not walkable or the square coordinates do not exist